BrainPortal/app/controllers/tasks_controller.rb

```diff
@@ -771,6 +771,395 @@ def apply_operation(operation, taskids, do_in_spawn = false, options = {})
+          next unless userfile.zenodo_deposit_id == combined_dep_id
+          userfile.zenodo_deposit_id = nil
+          userfile.save!
+          userfile.addlog(message)
+        end
+
+        flash.now[:notice] = message
+        combined_dep_id = zsite = deposit_id = nil # so we enter step 1 code below
+      else # Record DOI if needed
+        if @zenodo_deposit.submitted && @zenodo_deposit.metadata.doi.present?
+          zenodo_doi = @zenodo_deposit.metadata.doi
+          if @task.zenodo_doi.blank?
+            @task.update_column(:zenodo_doi, zenodo_doi)
+            @task.addlog("Zenodo DOI added: #{zenodo_doi}")
+          end
+          @zenodo_userfiles.compact.each do |userfile|
+            next if userfile.zenodo_doi.present?
+            userfile.update_column(:zenodo_doi, zenodo_doi)
+            userfile.addlog("Zenodo DOI added: #{zenodo_doi}")
+          end
+        end
+      end
+    end
+
+    # Step 1: ask the task for what we need to build a new zenodo deposit
+    # This sets up what's needed to render the initial form.
+    if combined_dep_id.blank?
+      @zenodo_deposit  = @task.base_zenodo_deposit
+      @zenodo_metadata = @zenodo_deposit.metadata || ZenodoClient::DepositMetadata.new
+
+      # Adjustments: creators
+      author = current_user.full_name.sub(/\s+(\S+)\s*$/,"")
+      last   = Regexp.last_match.try(:[],1)
+      author = "#{last}, #{author}" if last.present?
+      @zenodo_metadata.creators ||= []
+      @zenodo_metadata.creators.unshift(ZenodoClient::Author.new( :name => author )) unless
+        @zenodo_metadata.creators.any? { |a| a.name == author }
+    end
+
+    # All good for rendering the zenodo status page
+    return # renders zenodo.html.erb
+
+  end
+
+  # POST /tasks/:id/create_zenodo
+  def create_zenodo #:nodoc:
+    task_id = params[:id]
+    @task   = current_user.available_tasks.find(task_id)
+
+    if @task.zenodo_doi.present?
+      cb_error "A deposit has already been published.", :redirect => zenodo_task_path(@task)
+    elsif @task.zenodo_deposit_id.present?
+      cb_error "A deposit has already been created.", :redirect => zenodo_task_path(@task)
+    end
+
+    @zenodo_deposit     = ZenodoClient::Deposit.new(         zenodo_deposit_params.to_h          )
+    @zenodo_metadata    = ZenodoClient::DepositMetadata.new( zenodo_deposit_metadata_params.to_h )
+    zenodo_userfile_ids = @task.zenodo_outputfile_ids
+    @zenodo_userfiles   = zenodo_userfiles_from_ids(zenodo_userfile_ids)
+
+    # Adjustements: related files. All outputs that are already present
+    # with another DOI are just refered to as such.
+    related = @zenodo_userfiles
+      .select { |u| u.zenodo_doi.present? }
+      .map    { |u| ZenodoClient::RelatedIdentifier.new(:identifier => u.zenodo_doi, :relation => 'hasPart') }
+    if related.present?
+      @zenodo_metadata.related_identifiers ||= []
+      @zenodo_metadata.related_identifiers  += related
+    end
+
+    # Validate here; TODO
```

Comment:
Should be done or remove no ?

Reply:
Agreed. I'll add some validations, at least one.